src/__tests__/base.test.ts

```diff
@@ -55,26 +55,101 @@ describe("Base", () => {
     })
 
     it("inserts correct version header", async () => {
-      const json = await fetchPath(Service.CAR, "api/path")
+      const json = await fetchPath({ service: Service.CAR, path: "api/path" })
 
       expect(json).toEqual({ ok: true })
-      expect(fetch).toHaveBeenCalledWith(expect.any(String), {
-        headers: expect.objectContaining({
-          Accept: `application/vnd.carforyou.v1+json`,
-        }),
-      })
+      expect(fetch).toHaveBeenCalledWith(
+        expect.any(String),
+        expect.objectContaining({
+          headers: expect.objectContaining({
+            Accept: `application/vnd.carforyou.v1+json`,
+          }),
+        })
+      )
     })
 
     it("allows setting custom headers", async () => {
-      const json = await fetchPath(Service.CAR, "api/path", {
-        headers: { Foo: "bar" },
+      const json = await fetchPath({
+        service: Service.CAR,
+        path: "api/path",
+        options: {
+          headers: { Foo: "bar" },
+        },
       })
 
       expect(json).toEqual({ ok: true })
-      expect(fetch).toHaveBeenCalledWith(expect.any(String), {
-        headers: expect.objectContaining({
-          Foo: "bar",
-        }),
+      expect(fetch).toHaveBeenCalledWith(
+        expect.any(String),
+        expect.objectContaining({
+          headers: expect.objectContaining({
+            Foo: "bar",
+          }),
+        })
+      )
+    })
+
+    describe("access token handling", () => {
+      it("does not throw and error when the request is not marked as authorized", async () => {
+        const response = await fetchPath({
+          service: Service.CAR,
+          path: "api/path",
+          options: {
+            accessToken: "GIMME ACCESS!",
+          },
+        })
+        expect(response).toEqual({ ok: true })
+      })
```

Comment:
Shouldn't this test not include `accessToken` in the options?

Reply:
As in the way I understand it you `throw` when the request is marked as `isAuthorized` and there's no token, right?

Reply:
I can remove it if it makes it more clear, left it there on purpose as it should not matter weather you pass it or not for a public call...